### ISSUE TITLE
No notifications on macOS after build

### ISSUE THREAD

**Issue by kevinelliott** [mikaelbr/node-notifier#272]
I am getting notifications in my Electron app when it is in dev mode, but after it is built there are no notifications present.

I have my build setup with:

```
  node: {
    __filename: true,
    __dirname: true
  },
```

to account for `webpacker` and I also have my development and distribution profiles setup in Xcode which are detected during an Electron build. So I can (mostly) eliminate most of the common Electron and `electron-builder` issues that could cause `node-notifier` to be a problem.

Any thoughts here? I have no errors in the `Console` logging app.

What could be causing this?

**Comment by kevinelliott** [comment]
Any thoughts?

**Comment by Ivorfason** [comment]
the same here

**Comment by Ghefest** [comment]
the same problem on windows

**Comment by Noitidart** [comment]
Same problem here on macOS. When in dev mode it works fine, but when prod it doesnt' work.

It is working in dev mode even though I have:

```
  node: {
    __dirname: false,
    __filename: false
  },
```

**Comment by Aarbel** [comment]
same here

**Comment by Aarbel** [comment]
My package.json

```JSON
{
  …
  "main": "main.js",
  "scripts": {
    "postinstall": "electron-builder install-app-deps",
    "start": "electron main.js",
    "pack": "build --dir",
    "dist": "build"
  },
  "build": {
    "appId": "my.app.id",
    "dmg": {
      "contents": [
        {
          "x": 110,
          "y": 270
        },
        {
          "x": 370,
          "y": 270,
          "type": "link",
          "path": "/Applications"
        }
      ]
    },
    "mac": {
      "category": "productivity",
      "extendInfo": {
        "CFBundleURLName": "myapp",
        "CFBundleURLSchemes": [
          "myapp"
        ]
      }
    },
  },
  "dependencies": {
    "electron-is-dev": "^1.1.0",
    "electron-updater": "^4.0.6",
    "electron-window-state": "^5.0.3",
    "node-notifier": "^5.4.0",
    "url-parse": "^1.4.6"
  },
  "devDependencies": {
    "asar": "^1.0.0",
    "electron": "^4.1.4",
    "electron-builder": "^20.39.0",
    "electron-packager": "^13.1.1",
    "electron-reloader": "^0.2.0"
  }
}
```

**Comment by Aarbel** [comment]
@mikaelbr 

I tried with this instruction : https://github.com/mikaelbr/node-notifier#within-electron-packaging.

That generates very heavy .dmg and .zip files, and node-notifier still don't work.

**Comment by Aarbel** [comment]
After working one full day on node-notifier problems with Electron, i chose to use [node-mac-notifier](https://github.com/CharlieHess/node-mac-notifier) which works great under production.
`node-notifier` library seems powerful and has many stars, but many people seems to face build problems, even for windows.

As an example Slack use `node-mac-notifier` and `electron-windows-notifications` which are stable versions, even for very old versions of Electron. You don't have as many options but that's works.

**Comment by ReAlign** [comment]
You may need this: [node-notifier-in-electron-vue](https://github.com/ReAlign/node-notifier-in-electron-vue)

**Comment by Noitidart** [comment]
@ReAlign that's interesting, thanks for sharing. Even if wearen't using Vue though?

**Comment by ReAlign** [comment]
@Noitidart no, it‘s is just a simple handling of the situation **<No notifications on macOS after build>**,

I am just simple processing, ensure that can be used.

**Comment by pablopunk** [comment]
I fixed it with `"asar": false` in my [build config](https://github.com/pablopunk/serve-bar)

**Comment by leekangtaqi** [comment]
@Noitidart  add asarUnpack option to electron-builder

```
"build": {
    "asarUnpack": [
      "./node_modules/node-notifier/vendor/**"
    ],
    ...
}
```

**Comment by GyuCheol** [comment]
@leekangtaqi Thanks! You made my day!!
